status/api/provider.nim

```diff
@@ -23,3 +28,59 @@ proc callRpc*(self: StatusObject, rpcMethod: string, params: JsonNode):
     return CallRpcResult.ok await self.web3.callRpc(rpcMethod, params)
   except CatchableError as e:
     return CallRpcResult.err e.msg
+
+proc signTransaction(tr: var Transaction, pk: PrivateKey) =
+  let h = tr.txHashNoSignature
+  let s = sign(pk, SkMessage(h.data))
+
+  var r = toRaw(s)
+  let v = r[64]
+
+  tr.R = fromBytesBE(UInt256, r.toOpenArray(0, 31))
+  tr.S = fromBytesBE(UInt256, r.toOpenArray(32, 63))
+
+  tr.V = int64(v)
+
+  if tr.txType == TxType.TxLegacy:
+    tr.V += 27 # TODO! Complete this
```

Comment:
I wonder if we should move this to `status/private/callrpc.nim`? Maybe `callrpc.nim` isn't the best name 🤔

Reply:
Perhaps `status/api/provider.nim` and `status/private/callrpc.nim` could be renamed `status/api/ethereum_rpc.nim` and `status/private/ethereum_rpc.nim`, respectively?

I don't think it's critical for this PR, but definitely something good to think about.

If we wanted to be really forward-looking, we could have:
* `status/api/ethereum.nim`
* `status/api/ethereum/rpc.nim`
* `status/private/ethereum.nim`
* `status/private/ethereum/rpc.nim`

What I mean is that nim-status might embed [fluffy](https://our.status.im/nimbus-fluffly/) at some point, in which case the library's users could choose to `import status/api/ethereum/fluffy` and use direct calls, or instead choose to `import status/api/ethereum/rpc` and make calls to some endpoint, e.g. an Infura endpoint or a geth node they're running on their home network.